ControlMeasurements/Controllers/HomeController.cs

```diff
@@ -36,9 +36,109 @@ public async Task<IActionResult> Create([Bind("Place,Measurement")] WaterMeasure
                 _context.Add(water);
                 await _context.SaveChangesAsync();
                 return RedirectToAction(nameof(Index));
-            }   
+            }
             return View(water);
         }
+        public async Task<IActionResult> Edit(Guid? id)
+        {
+            if (id == null)
+            {
+                return NotFound();
+            }
+
+            var waterMeasurement = await _context.WaterMeasurements.SingleOrDefaultAsync(m => m.Id == id);
+            if (waterMeasurement == null)
+            {
+                return NotFound();
+            }
+            return View(waterMeasurement);
+        }
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Edit(Guid Id, [Bind("Id,Place,Measurement")] WaterMeasurement waterMeasurement)
+        {
+            if (Id != waterMeasurement.Id)
+            {
+                return NotFound();
+            }
+
+            if (ModelState.IsValid)
+            {
+                try
+                {
+                    _context.Update(waterMeasurement);
+                    await _context.SaveChangesAsync();  
+                }
+                catch (DbUpdateConcurrencyException)
+                {
+                    if (!WaterMeasurement(waterMeasurement.Id))
+                    {
+                        return NotFound();
+                    }
+                    else
+                    {
+                        throw;
+                    }
+                }
+                return RedirectToAction(nameof(Index));
+            }
+            return View(waterMeasurement);
+            
+        }
+
+        private bool WaterMeasurement(Guid id)
```

Comment:
Z kodu powyżej wynika, że ta metoda powinna sprawdzać czy wpis o danym id istnieje, prawda? Aktualnie w środku mamy NotImplementedException, więc jeśli kiedykolwiek tam wejdzie to się wysypie :)

Reply:
Jeśli chcesz używać tego sprawdzenia, to musisz to zaimplementować. I najlepiej zmienić na nazwę metody na _Exists(Guid id)_ czy cos takiego :)

Reply:
Poprawione :)